.github/workflows/validate-pull-request.yaml

```diff
@@ -0,0 +1,20 @@
+name: PR Validation
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v7
+        with:
+          java-version: graalvm@20.0.0
+      - name: Check formatting
+        run: sbt scalafmtCheckAll
+      - name: Compile
+        run: sbt Compile/compile Test/compile
```

Comment:
This is a good idea, hopefully this will help us catch a few bugs before having other people look at the PR.